src/main/java/org/jpeek/graph/Disjoint.java

```diff
@@ -48,6 +50,26 @@ public Disjoint(final Graph graph) {
 
     @Override
     public List<Set<Node>> value() throws Exception {
-        throw new UnsupportedOperationException(this.graph.toString());
+        final Set<Node> unvisited = new HashSet<>(this.graph.nodes());
+        final List<Set<Node>> result = new ArrayList<>(unvisited.size());
+        while (!unvisited.isEmpty()) {
```

Comment:
@HDouss Sorry, I could catch the whole algorithm, but is it possible to use `iterator.hasNext()` for the `unvisited`?

Reply:
@fanifieiev it is not really possible, unless to get a new iterator from the set in each iteration. This set is getting shrinked (by more than one element) in each iteration. Even the element at which the iterator is pointing is being removed from the set.